spec/lib/services/nodes-api-service-spec.js

```diff
@@ -501,6 +516,148 @@ describe("Http.Services.Api.Nodes", function () {
                                              {relations: enclRelationAfter});
             });
         });
+
+        it("should delete component nodes", function() {
+            this.sandbox.stub(nodeApiService, 'removeNode').resolves();
+            return nodeApiService._removeRelation(
+                computeNode, 'enclosedBy', "1234abcd1234abcd1234abcf"
+            ).then(function() {
+                expect(nodeApiService.removeNode).to.be.calledWithExactly(
+                    computeNode, 'enclosedBy'
+                );
+            });
+        });
+    });
+
+    describe("_addRelation", function() {
+        var computeNode2;
+        beforeEach(function() {
+            computeNode2 = {
+                id: '1234abcd1234abcd1234abcx',
+                type: 'compute',
+                relations: [
+                    {
+                        "relationType": "enclosedBy",
+                        "targets": [
+                            "1234abcd1234abcd1234abcf"
+                        ]
+                    }
+                ]
+            };
+            updateByIdentifier.resolves();
+        });
+
+        it("should do nothing if arguments are missing", function() {
+            var argList = [rackNode, "contains", [computeNode]];
+            _.forEach(argList, function(arg, index) {
+                var argsCopy = [].concat(argList);
+                argsCopy[index] = undefined;
+                nodeApiService._addRelation(argsCopy[0], argsCopy[1], argsCopy[2])
+                .then(function() {
+                    expect(updateByIdentifier).to.not.be.called;
+                });
+            });
+        });
+
+        it("should return a promise for an update to the relations field of a node", function() {
+            return nodeApiService._addRelation(
+                rackNode, 'contains', [computeNode.id, computeNode2.id]
+            ).then(function() {
+                expect(updateByIdentifier).to.be.calledWithExactly(rackNode.id, {relations:
+                    [{relationType: 'contains', targets: [computeNode.id, computeNode2.id]}]
+                });
+            }).then(function() {
+                rackNode.relations = [{relationType: 'contains', targets: [computeNode.id]}];
+                return nodeApiService._addRelation(rackNode, 'contains', [computeNode2.id]);
+            }).then(function() {
+                expect(updateByIdentifier).to.be.calledWithExactly(rackNode.id, {relations:
```

Comment:
updateByIdentifier need be reset before the 2nd `_addRelation`. Otherwise, it only reflects the result of first call of `_addRelation`.